### ISSUE TITLE
When parsing a `StaticPolicySet` from a text representation, only the first error gets recorded. 

### ISSUE THREAD

**Issue by AndreasKoestler** [cedar-policy/cedar#1293]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/cedar-policy/cedar/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/cedar-policy/cedar/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### Bug Category

Cedar Parser

### Describe the bug

When parsing a `StaticPolicySet` from a text representation, only the first error gets recorded. This makes handling the parsing results difficult. The use case at hand is the integration with a code editor - in general you want to handle/hint on all parsing errors for a given policy set.  

### Expected behavior

All errors in a provided text representation of a ``StaticPolicySet` should be returned to allow processing of the errors, e.g. displaying errors in a code editor.

### Reproduction steps

Call `check_parse_policy_set` on the text representation of a policy set containing more than one error.
  

### Code Snippet

```rust
let content = r#"permit(
  principal == U ser::"alice", 
  action    == Act ion::"view", 
  reso urce  == Photo::"VacationPhoto94.jpg"
); // We have three syntax errors here. 
"#;
let maybe_policies = PolicySet.from_str(content);
let res = match maybe_policies {
        Ok(policies) => check_parse_policy_set(policies), // This should report 3 errors but only reports the first. 
        Err(err) => todo!()
};

let err_count = match result {
        CheckParseAnswer::Success => 0,
        CheckParseAnswer::Failure { errors } => errors.len(),
};

assert_eq!(err_count, 3);
```

### Log output

```
// Put your output below this line

```

### Operating  System

Linux

**Comment by john-h-kastner-aws** [comment]
Hi, thanks for the question. Our pares error messages include a `related` field which contain some other associated error messages. If you look at that list of errors you should see that everything is reported. These FFI interfaces aren't the most well documented, but here's a link to it in the docs: https://docs.rs/cedar-policy/latest/cedar_policy/ffi/struct.DetailedError.html#structfield.related

**Comment by AndreasKoestler** [comment]
Indeed, `related` contains the remaining errors. For the above example, this yields:
```
Top level: "failed to parse policies from string: unexpected token `ser`"
related: Array(2)
0: 'unexpected token `ion`
1: 'unexpected token `to`'
```
However, this breaks down if I modify the example slightly:
```
permit(
  prin cipal == U ser::"alice", 
  action    == Act ion::"view", 
  resource  == Pho to::"VacationPhoto94.jpg"
)
```
Now I only get the top-level error and no related errors:
```
failed to parse policies from string: unexpected token `cipal`
```
Is this an issue with parsing and error reporting or am I not using the parsing interface correctly?

**Comment by john-h-kastner-aws** [comment]
Now you're running into a known limitation in how our parser handles error recovery. When there's a parse error in that position, it skips ahead to the start of the next _policy_, ignoring any other parse error which may exist in the current policy. For the other errors it is able to recover and continue parsing within the same policy. This could be improved, but likely won't be a priority for us to get to.

If you were interested in hacking on our parsers and contributing an improvement yourself, the place to start would be the [relevant production in our grammar](https://github.com/cedar-policy/cedar/blob/887a85ad169702f3f3fee7db3188bdafdbff80fd/cedar-policy-core/src/parser/grammar.lalrpop#L121) and the [lalrpop documentation for error recovery](https://lalrpop.github.io/lalrpop/tutorial/008_error_recovery.html).

You might also run into the related issue #1054 where a different category of error will only be reported for one policy even though it could be reported for multiple at a time.
